app/src/main/java/com/gathering/android/auth/password/ForgetPasswordScreen.kt

```diff
@@ -26,40 +56,103 @@ class ForgetPasswordScreen : FullScreenBottomSheet(), ForgetPasswordNavigator {
         super.onViewCreated(view, savedInstanceState)
 
-        lifecycleScope.launch {
-            viewModel.uiState.collectLatest { state ->
-                if (state.isInProgress) {
-                    binding.btnSendLink.startAnimation()
-                } else {
-                    binding.btnSendLink.revertAnimation()
-                }
-                state.errorMessage?.let {
-                    showErrorText(it)
-                }
+        if (isComposeEnabled) {
+            viewModel.onViewCreated(this)
+            return
+        } else {
+            lifecycleScope.launch {
+                viewModel.uiState.collectLatest { state ->
+                    if (state.isInProgress) {
+                        binding.btnSendLink.startAnimation()
+                    } else {
+                        binding.btnSendLink.revertAnimation()
+                    }
+                    state.errorMessage?.let {
+                        showErrorText(it)
+                    }
 
+                }
             }
-
-        }
-
-        binding.btnSendLink.setOnClickListener {
-            val email = binding.etEmailInput.text.toString()
-            viewModel.onSendLinkBtnClicked(email)
+            binding.btnSendLink.setOnClickListener {
+                val email = binding.etEmailInput.text.toString()
+                viewModel.onSendLinkBtnClicked(email)
+            }
+            viewModel.onViewCreated(this)
         }
-
-        viewModel.onViewCreated(this)
-
     }
 
     override fun navigateToResetPassInfoBottomSheet() {
         findNavController().navigate(
             R.id.action_forgetPasswordScreen_to_resetPassInfoBottomSheet
         )
     }
+
+    @Composable
+    @Preview(showBackground = true)
+    fun ForgetPasswordScreenPreview() {
+        ForgetPasswordScreenCompose(isInProgress = true, error = "error")
+    }
+
+    @Composable
+    fun ForgetPasswordScreenCompose(isInProgress: Boolean, error: String? = null) {
+        var email by rememberSaveable { mutableStateOf("") }
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .padding(10.dp),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            CustomTextView(
+                textResId = R.string.reset_pass_msg,
+                modifier = Modifier
+                    .fillMaxWidth(),
+                textStyle = TextStyle(
+                    fontWeight = FontWeight.Bold, fontSize = 18.sp, color = Color.Black
+                )
+            )
+            GatheringEmailTextField(
+                value = email,
+                onValueChange = { email = it },
+                label = "Email"
+            )
+                CustomActionButton(
+                    text = "SEND LINK", onClick = {
+                        viewModel.onSendLinkBtnClicked(email)
```

Comment:
I suggest to not call the view model here. Pass a click higher order function as the parameter of your Composable, in this way you make it more reusable and actually a composable, in this way your composable is highly tied to only this screen.

Reply:
If we have the same pattern somewhere else we can improve it by doing the same.

Reply:
lets extract any click listener and pass it in as a parameter everywhere. you can do it in a separate ticket too. @idaosk would you please create a ticket for it ?

Reply:
@amirziarati https://trello.com/c/iMVRX96h/132-extract-any-click-listener-and-pass-it-in-as-a-parameter

Reply:
lets merge this for now and get back to that and fix it later.
we need to dig more into best practices and find some other rules like this to create a convention.

Reply:
ill let Mo approve it if he is ok

Reply:
@moyarsuccess can you approve for now? I created a seperate ticket to fix this later because there are multiple classes that need to be fixed !!
 https://trello.com/c/iMVRX96h/132-extract-any-click-listener-and-pass-it-in-as-a-parameter

Reply:
i also think we should change viewmodel to be a method level variable rather than a field when we got rid of the feature flag too. this way we cant go wrong using that in sub-composes

Reply:
Shit, how you could edit my message fuck

Reply:
You can merge it yes, Definity

Reply:
if you guys are done playing, can someone approve this PR? :))))

Reply:
> ill let Mo approve it if he is ok, Fuck you Amir

nice comeback :)))))))))))